chiselFrontend/src/main/scala/chisel3/package.scala

```diff
@@ -201,91 +138,17 @@ package object chisel3 {    // scalastyle:ignore package.object.name
   /** This contains literal constructor factory methods that are deprecated as of Chisel3.
     * These will be removed very soon. It's recommended you port your code ASAP.
     */
-  trait UIntFactory extends UIntFactoryBase {
-    /** Create a UInt literal with inferred width. */
-    @chiselRuntimeDeprecated
-    @deprecated("use n.U", "chisel3, will be removed by end of 2017")
-    def apply(n: String): UInt = n.asUInt
-
-    /** Create a UInt literal with fixed width. */
-    @chiselRuntimeDeprecated
-    @deprecated("use n.U(width.W)", "chisel3, will be removed by end of 2017")
-    def apply(n: String, width: Int): UInt = n.asUInt(width.W)
-
-    /** Create a UInt literal with specified width. */
-    @chiselRuntimeDeprecated
-    @deprecated("use value.U(width)", "chisel3, will be removed by end of 2017")
-    def apply(value: BigInt, width: Width): UInt = value.asUInt(width)
-
-    /** Create a UInt literal with fixed width. */
-    @chiselRuntimeDeprecated
-    @deprecated("use value.U(width.W)", "chisel3, will be removed by end of 2017")
-    def apply(value: BigInt, width: Int): UInt = value.asUInt(width.W)
-
-    /** Create a UInt literal with inferred width.- compatibility with Chisel2. */
-    @chiselRuntimeDeprecated
-    @deprecated("use value.U", "chisel3, will be removed by end of 2017")
-    def apply(value: BigInt): UInt = value.asUInt
-
-    /** Create a UInt with a specified width */
-    @chiselRuntimeDeprecated
-    @deprecated("use UInt(width.W)", "chisel3, will be removed by end of 2017")
-    def width(width: Int): UInt = apply(width.W)
-
-    /** Create a UInt port with specified width. */
-    @chiselRuntimeDeprecated
-    @deprecated("use UInt(width)", "chisel3, will be removed by end of 2017")
-    def width(width: Width): UInt = apply(width)
-  }
+  trait UIntFactory extends UIntFactoryBase
```

Comment:
Can this trait be eliminated? It smells like an intermediate class that doesn't do anything anymore?

Reply:
Yeah, these can be flattened. Updated in the rebase to rename `trait UIntFactoryBase` to `trait UIntFactory`, remove `UIntFactory`, and make anything that extended `UIntFactoryBase` now extend `UIntFactory`.